### ISSUE TITLE
Placing lifetime specifier after `&mut` emits wall of unhelpful diagnostics

### ISSUE THREAD

**Issue by SNCPlay42** [rust-lang/rust#73568]
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=37d6d3054c9458501aeaa77c20000e8c)

```rust
fn x<'a>(x: &mut 'a i32){}
```
Expected result: a diagnostic suggests placing `'a` before `mut`

Actual diagnostic output:
<details>

```
   Compiling playground v0.0.1 (/playground)
error: lifetime in trait object type must be followed by `+`
 --> src/lib.rs:1:18
  |
1 | fn x<'a>(x: &mut 'a i32){}
  |                  ^^

error: expected one of `:`, `@`, or `|`, found `)`
 --> src/lib.rs:1:24
  |
1 | fn x<'a>(x: &mut 'a i32){}
  |                        ^ expected one of `:`, `@`, or `|`
  |
  = note: anonymous parameters are removed in the 2018 edition (see RFC 1685)
help: if this was a parameter name, give it a type
  |
1 | fn x<'a>(x: &mut 'a i32: TypeName){}
  |                     ^^^^^^^^^^^^^
help: if this is a type, explicitly ignore the parameter name
  |
1 | fn x<'a>(x: &mut 'a _: i32){}
  |                     ^^^^^^

error: expected one of `)` or `,`, found `i32`
 --> src/lib.rs:1:21
  |
1 | fn x<'a>(x: &mut 'a i32){}
  |                    -^^^ expected one of `)` or `,`
  |                    |
  |                    help: missing `,`

warning: trait objects without an explicit `dyn` are deprecated
 --> src/lib.rs:1:18
  |
1 | fn x<'a>(x: &mut 'a i32){}
  |                  ^^ help: use `dyn`: `dyn 'a`
  |
  = note: `#[warn(bare_trait_objects)]` on by default

error[E0224]: at least one trait is required for an object type
 --> src/lib.rs:1:18
  |
1 | fn x<'a>(x: &mut 'a i32){}
  |                  ^^

error: aborting due to 4 previous errors; 1 warning emitted

For more information about this error, try `rustc --explain E0224`.
error: could not compile `playground`.

To learn more, run the command again with --verbose.
```
</details>

**Comment by SNCPlay42** [comment]
Output from rustc 1.43:

```
error: expected type, found `'a`
 --> <source>:1:18
  |
1 | fn x<'a>(x: &mut 'a i32){}
  |                  ^^ expected type

error: aborting due to previous error
```

Seems to have regressed recently.

**Comment by SNCPlay42** [comment]
`cargo-bisect-rustc`:

searched nightlies: from nightly-2020-03-09 to nightly-2020-06-21
regressed nightly: nightly-2020-03-12
searched commits: from https://github.com/rust-lang/rust/commit/15812785344d913d779d9738fe3cca8de56f71d5 to https://github.com/rust-lang/rust/commit/c20d7eecbc0928b57da8fe30b2ef8528e2bdd5be
regressed commit: https://github.com/rust-lang/rust/commit/c20d7eecbc0928b57da8fe30b2ef8528e2bdd5be

<details>
<summary>bisected with <a href='https://github.com/rust-lang/cargo-bisect-rustc'>cargo-bisect-rustc</a> v0.5.2</summary>


Host triple: x86_64-unknown-linux-gnu
Reproduce with:
```bash
cargo bisect-rustc --prompt --start 2020-03-09 
```
</details>

Looks like #69760?

**Comment by SNCPlay42** [comment]
Further testing confirms #69760, specifically [`ba3ae46`](https://github.com/rust-lang/rust/pull/69760/commits/ba3ae46de972f3e70319c25a5da69378454cb88d) introduced the large number of diagnostics.

**Comment by LeSeulArtichaut** [comment]
Assigning `P-high` [as discussed as part of the Prioritization WG procedure](https://rust-lang.zulipchat.com/#narrow/stream/227806-t-compiler.2Fwg-prioritization/topic/I-prioritize.20.2373568.20Placing.20lifetime.20specifier.20after.20.60.26mut.60).
